core/arch/arm/arm.mk

```diff
@@ -145,8 +145,14 @@ $(call force,CFG_DEBUG_INFO,y)
 endif
 
 CFG_CC_OPTIMIZE_FOR_SIZE ?= y
+CFG_CC_OPTIMIZE_FOR_PERFORMANCE ?= n
+
+$(call cfg-check-at-most-one,CFG_CC_OPTIMIZE_FOR_SIZE CFG_CC_OPTIMIZE_FOR_PERFORMANCE)
```

Comment:
LGTM.

This means building for perf needs the 2 configs set:
`make CFG_CC_OPTIMIZE_FOR_PERFORMANCE=y CFG_CC_OPTIMIZE_FOR_SIZE=n ...`


Could add
```
CFG_CC_OPTIMIZE_FOR_PERFORMANCE ? = n
ifeq ($(CFG_CC_OPTIMIZE_FOR_PERFORMANCE),y)
CFG_CC_OPTIMIZE_FOR_SIZE ?= n
else
CFG_CC_OPTIMIZE_FOR_SIZE ?= y
endif
```
but looks not clean.

Reply:
Yeah, and things would get even uglier with one more optimization level. For instance if we were to add `CFG_CC_OPTIMIZE_FOR_PERFORMANCE_O3`. The linux kernel has such a config, I am using a similar scheme here. But Linux has Kconfig so the constraints and dependencies are better handled.

I have a better proposal I think: https://github.com/OP-TEE/optee_os/pull/4073

Reply:
https://github.com/OP-TEE/optee_os/pull/4073 looks better yes.